homeassistant/components/sensor/yweather.py

```diff
@@ -0,0 +1,196 @@
+"""
+Support for the Yahoo! Weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.yweather/
+"""
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.const import (CONF_PLATFORM, TEMP_CELSIUS, TEMP_FAHRENHEIT,
+                                 CONF_MONITORED_CONDITIONS, STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ["yahooweather==0.3"]
+
+YAHOO_WEATHER = {}
+SENSOR_TYPES = {
+    'weather_current': ['Current', None],
+    'weather': ['Condition', None],
+    'temperature': ['Temperature', "temperature"],
+    'temp_min': ['Temperature', "temperature"],
+    'temp_max': ['Temperature', "temperature"],
+    'wind_speed': ['Wind speed', "speed"],
+    'humidity': ['Humidity', "%"],
+    'pressure': ['Pressure', "pressure"],
+    'visibility': ['Visibility', "distance"],
+}
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): "yweather",
+    vol.Required("woeid"): vol.Coerce(str),
+    vol.Optional("forecast", default=0): vol.Coerce(int),
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        [vol.In(SENSOR_TYPES.keys())],
+})
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yahoo! weather sensor."""
+    from yahooweather import UNIT_C, UNIT_F
+
+    unit = hass.config.temperature_unit
+    woeid = config.get("woeid", None)
+    forecast = config.get("forecast", 0)
+
+    if woeid is None:
+        _LOGGER.critical("You need woeid from yahoo!")
+        return False
+
+    if forecast > 4:
```

Comment:
I guess that we could catch this in the schema too. But a more descriptive log entry is nice too. Wasn't there much more forecast available in the past, like 10 days?